orchagent/routeorch.cpp

```diff
@@ -18,18 +22,33 @@ RouteOrch::RouteOrch(DBConnector *db, string tableName, NeighOrch *neighOrch) :
 {
     SWSS_LOG_ENTER();
 
+    sai_attribute_t attr;
+    attr.id = SAI_SWITCH_ATTR_NUMBER_OF_ECMP_GROUPS;
+
+    sai_status_t status = sai_switch_api->get_switch_attribute(1, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_WARN("Failed to get switch attribute number of ECMP groups. \
+                       Use default value. rv:%d", status);
+        m_maxNextHopGroupCount = DEFAULT_NUMBER_OF_ECMP_GROUPS;
```

Comment:
looks like different vendor has different sai interpretation of this attribute, let's do a vendor specific implementation.